Change config location to /etc/dunst/dunstrc

fwSmit opened the pull request:
Fixes #795 
The makefile still isn't fully xdg compliant. If the user has set `$XDG_CONFIG_DIRS` it should be reflected our install, I believe.
I didn't know the best way to implement that, but [this](https://www.gnu.org/software/make/manual/make.html#Directory-Variables) is some make documentation about all the standard directories, and [this](https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html) is the spec.

fwSmit commented:
Weird, this commit doesn't seem to cause the ci failing. My bisect points to 58d215ddfeaf3fae9f017d05b8c886d7f4aadffd.

tsipinakis commented:
> that's only on wayland and is not a problem. I can't find what causes the ci failing

The CI is failing in the `test-install` script. See [here](https://github.com/dunst-project/dunst/runs/1605771381#step:8:16).

Re: This PR. I'm thinking in Debian terms where it's usual to have docs and example config files in `/usr/share/doc/<pkg>` and it's now muscle memory to search there when I want something. Looking in `/etc/xdg` is counter-intuitive in that sense. So, I'd opt for having both, one in `/usr` for documentation and the `/etc/xdg` one for actually loading it.

fwSmit commented:
> > that's only on wayland and is not a problem. I can't find what causes the ci failing
> 
> The CI is failing in the `test-install` script. See [here](https://github.com/dunst-project/dunst/runs/1605771381#step:8:16).

Ah thanks. To fix it I need to make sure the config is installed somewhere else when the prefix is changed. Do I need to make a new CONFIG_PREFIX or change PREFIX to `/` ?


> Re: This PR. I'm thinking in Debian terms where it's usual to have docs and example config files in `/usr/share/doc/<pkg>` and it's now muscle memory to search there when I want something. Looking in `/etc/xdg` is counter-intuitive in that sense. So, I'd opt for having both, one in `/usr` for documentation and the `/etc/xdg` one for actually loading it.

I don't think it's a good idea to place two identical files in different places. Linking them doesn't seem good either. Maybe this is too distro specific. On arch I'm used to the config being something like `/etc/dunst/dunstrc`. There are only a few configs in `/etc/xdg/...`. I'm interested what other people think is the best place for the config.
We can also just make dunst search `/usr/share/doc/dunst/dunstrc`, but I think that's also a bad solution.

tsipinakis commented:
Linking or making dunst look in /usr/share/doc isn't going to work in Debian at least as everything in that directory is gzipped. Debian-wise I'd always opt to install the config file in both locations as the `/usr/share/doc` directory is muscle memory to all users by now, (and Debian policy suggests it).

The question here is what's best for upstream, default behaviour, does this extend to other distros?

> Do I need to make a new CONFIG_PREFIX or change PREFIX to / ?

`DESTDIR` should be the target 'root' so changing `PREFIX` would break convention here. I'm not sure what the standard variable for `/etc/` files is, if there is one. We should check what other projects do with such files.

fwSmit commented:
> Linking or making dunst look in /usr/share/doc isn't going to work in Debian at least as everything in that directory is gzipped. Debian-wise I'd always opt to install the config file in both locations as the `/usr/share/doc` directory is muscle memory to all users by now, (and Debian policy suggests it).

[This](https://www.debian.org/doc/manuals/maint-guide/dother.en.html#conffiles) page suggests that conffiles should be under /etc for debian. Every other distro I can find installs it's dotfiles in /etc as well.

> The question here is what's best for upstream, default behaviour, does this extend to other distros?

I think it might be best to install them to `/etc/dunst/dunstrc` and change dunst to look there as well.

> `DESTDIR` should be the target 'root' so changing `PREFIX` would break convention here. I'm not sure what the standard variable for `/etc/` files is, if there is one. We should check what other projects do with such files.

Oh thanks, that's what I was looking for. I changed the test-install script to use that instead of PREFIX. Also made some other improvements to it.

tsipinakis commented:
> This page suggests that conffiles should be under /etc for debian

I guess the divide here is that I view the dunstrc as both a configuration file (which is what this change is about) but also a big part of dunsts limited documentation/examples. Which is why I hesitate moving it entirely off that directory.

> I think it might be best to install them to /etc/dunst/dunstrc and change dunst to look there as well.

That might be a good compromise, my holdout is that `/etc/xdg` is a rarely used path (no other program on my system has any config there other than autostart files).

Also, don't forget to update the path in dunsts manpage.

fwSmit commented:
I finished this PR and rebased onto the `wayland-2` branch. There were no conflicts.

tsipinakis commented:
> I finished this PR and rebased onto the wayland-2 branch. There were no conflicts.

Apparently the merge commit confuses it. Both PRs have conflicts.

fwSmit commented:
> > I finished this PR and rebased onto the wayland-2 branch. There were no conflicts.
> 
> Apparently the merge commit confuses it. Both PRs have conflicts.

No that's because I added some stuff to wayland-2 because of bebehei's comments. I rebased again and fixed the conflicts now.
I noticed that the changes from #793 also show up in the "files changed" tab. I don't know why, I just rebased on upstream/master.

fwSmit commented:
Could this be reviewed? Then it can be merged together with #799. Otherwise we will stack up conflicts.

tsipinakis commented:
Sorry, took a look at this yesterday and forgot to comment. This needs to be rebased. Merge commit is d3dd26a but in mater it's d3dd26a. I feel like this will cause some issues (or at least confuse history) if I merge as-is.

fwSmit commented:
Yeah I thought so, but didn't know how to fix. I rebased again and it has been cleaned up

tsipinakis reviewed (approved):
LGTM.

While reviewing this I couldn't help but think that we should simplify the `test-install` script to only provide the `DESTDIR` variable and let everything else be the default. But that's out of scope of this.

fwSmit commented:
> While reviewing this I couldn't help but think that we should simplify the `test-install` script to only provide the `DESTDIR` variable and let everything else be the default. But that's out of scope of this.

I don't think that's better. Setting these variables to some non-default value makes sure you aren't hardcoding the defaults in the makefile.

I noticed btw that I didn't update commit message. It should be "Change config location to /etc/dunst/dunstrc". Well I guess it's too late to change it now.
